neo.UnitTests/UT_ECDsa.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Security.Cryptography;
+using System.Numerics;
+
+namespace Neo.Cryptography.ECC.Tests
+{
+    [TestClass()]
+    public class UT_ECDsa
+    {
+        private ECDsa ecdsa;
+
+        public static byte[] generatekey(int privateKeyLength)
+        {
+            byte[] privateKey = new byte[privateKeyLength];
+            using (RandomNumberGenerator rng = RandomNumberGenerator.Create())
+            {
+                rng.GetBytes(privateKey);
+            }
+            return privateKey;
+        }
+
+        [TestMethod()]
+        public void KeyRecoverTest()
+        {
+            byte[] privateKey = generatekey(32);
+            ECPoint publickey = ECCurve.Secp256k1.G * privateKey;
+            ecdsa = new ECDsa(privateKey, ECCurve.Secp256k1);
+            byte[] message = System.Text.Encoding.Default.GetBytes("HelloWorld");
+            BigInteger[] signatures = ecdsa.GenerateSignature(message);
+            BigInteger r = signatures[0];
+            BigInteger s = signatures[1];
+            bool v;
+            if (signatures[2] == 0)
+            {
+                v = true;
+            }
+            else
+            {
+                v = false;
+            }
+            ECPoint recoverKey = ECDsa.KeyRecover(ECCurve.Secp256k1, r, s, message, v, true);
+            Assert.IsTrue(recoverKey.Equals(publickey));
+            //wrong key part
+            r = new System.Numerics.BigInteger(generatekey(32));
+            try
+            {
+                recoverKey = ECDsa.KeyRecover(ECCurve.Secp256k1, r, signatures[1], message, true, true);
+                Assert.IsFalse(recoverKey.Equals(publickey));
```

Comment:
With this `try` you never know if the test works or not. Please, hardcode the privateKey, and remove the try.

Reply:
Right, this part is wrong key recover test, sometimes random "r" can raise expcetion in decompress point, I would like to add exception type judgement to solve that.